buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -83,6 +83,19 @@ class BuildPlugin implements Plugin<Project> {
 
     /** Adds additional manifest info to jars */
     static void configureJarManifest(Project project) {
+        project.scminfo {
+            def dotgit = "$project.rootProject.projectDir/.git/"
+            def head = new File(dotgit + "HEAD").text.split(":")
```

Comment:
I've been preferring to actually use types vs `def`. I don't think there is any advantage to def, and why not when we know the types anyways? It makes the code easier to read too.